### PR TITLE
Updated Game to split it into states based on pages

### PR DESCRIPTION
**Pull request by hmeister10**
GAME
 -> Lobby
 -> Question

### CODE REVIEW

File: api/answer.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { httpClient } from "./httpClient";
 
-export const sendAnswer = async (answerId) => {
+export const voteForAnswer = async (answerId) => {
```
Comment (neha-gehani) [inline api/answer.ts]: specify type data. Check the other services for ref

File: api/interface.ts [hunk 2]
```diff
@@ -10,10 +10,25 @@ export interface Participant {
   name: string;
 }
 
+export interface Answer {
+  id: number;
+  answerStr: string;
+  user?: User
```
Comment (neha-gehani) [inline api/interface.ts]: missing ;

File: api/interface.ts [hunk 3]
```diff
@@ -10,10 +10,25 @@ export interface Participant {
...
+}
+
+export interface Question {
+  id: number;
+  questionStr: string;
+  status: string;
+  answer?: Answer[]
```
Comment (neha-gehani) [inline api/interface.ts]: missing ;

File: api/interface.ts [hunk 4]
```diff
@@ -10,10 +10,25 @@ export interface Participant {
...
+}
+
 export interface Game {
+  [x: string]: any;
```
Comment (neha-gehani) [inline api/interface.ts]: what?

File: api/questions.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import { httpClient } from "./httpClient";
 
-export const sendAnswer = async (questionId, answer) => {
-  const result = await httpClient.post(`/quesiton/${questionId}/answer`, {answer});
+export const sendAnswer = async (questionId, answerStr) => {
```
Comment (neha-gehani) [inline api/questions.ts]: Specify type data

File: pages/game/[gameId].tsx [hunk 6]
```diff
@@ -6,39 +6,83 @@ import { getGame, startGame } from "../../api/game";
 import { LayoutPageProps } from "../_app";
 import { getUser } from "../../api/user";
 import GameNotStarted from "../../components/game/GameNotStarted";
+import GameQuestion from "../../components/game/GameQuestion";
 import GameCode from "../../components/game/GameCode";
-import { isGameStarted } from "../../helpers/game";
-import { useRouter } from "next/router";
 
 interface GamePageProps extends LayoutPageProps {
+  gameId: String;
```
Comment (neha-gehani) [inline pages/game/[gameId].tsx]: why? game id is a part of the game object

File: api/interface.ts (hunk 4 above)
Comment (hmeister10) [reply api/interface.ts]: this was u only, ill change it

**Review by neha-gehani — APPROVED** [review]

**Pull request merged**